staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -232,7 +235,9 @@ func NewReflectorWithOptions(lw ListerWatcher, expectedType interface{}, store S
 		minWatchTimeout = options.MinWatchTimeout
 	}
 	r := &Reflector{
-		name:            options.Name,
+		name: options.Name,
+		// we need this to be unique per process (some names are still the same)but obvious who it belongs to
+		metrics:         newReflectorMetrics(makeValidPromethusMetricName(fmt.Sprintf("reflector_"+options.Name+"_expectedType_"+reflect.TypeOf(expectedType).String()+"_%07d", rand.Intn(1000000)))),
```

Comment:
won't the randomized suffix explode metrics cardinality?

Reply:
The randomized suffix can cause metric cardinality explosion, worsening Prometheus' storage and query performance. The randomized suffix can be removed here.

Reply:
@liggitt done